src/main/java/org/openstreetmap/atlas/streaming/resource/ResourceCloseable.java

```diff
@@ -0,0 +1,48 @@
+package org.openstreetmap.atlas.streaming.resource;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Use when the returned resource has some items that cannot be closed prior to finishing with this
+ * resource.
+ *
+ * @author Taylor Smock
+ */
+public interface ResourceCloseable extends Resource, AutoCloseable
+{
+    @Override
+    default void close() throws Exception
+    {
+        final Collection<Exception> exceptions = new ArrayList<>();
+        for (final AutoCloseable closeable : Stream.of(getDependencies()).filter(Objects::nonNull)
+                .collect(Collectors.toList()))
+        {
+            try
+            {
+                closeable.close();
+            }
+            catch (final Exception exception)
+            {
+                exceptions.add(exception);
+            }
+        }
+        if (!exceptions.isEmpty())
+        {
+            throw exceptions.iterator().next();
```

Comment:
It would be nice here to have a more detailed message, with the number of exceptions, and the cause set as the first one?
```java
throw new CoreException("{} exceptions thrown while closing {}", 
    exceptions.size(), this.getName(), exceptions.iterator().next());
```

Reply:
This can be done. I really wanted to merge them all together, initially, but I ended up deciding that it might be a good idea to just throw the first once, since later exceptions might be thrown due to the first exception.